src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyEstimator.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Transforms.TimeSeries
+{
+    /// <summary>
+    /// Detect timeseries anomalies for entire input using Spectral Residual(SR) algorithm.
+    /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    /// To create this estimator, use
+    /// [DetectEntireAnomalyBySrCnn](xref:Microsoft.ML.TimeSeriesCatalog.DetectEntireAnomalyBySrCnn(Microsoft.ML.TransformsCatalog,System.String,System.String,System.String,System.Double,System.Int32,SrCnnDetectMode,System.Double))
+    /// ###  Estimator Characteristics
+    /// |  |  |
+    /// | -- | -- |
+    /// | Does this estimator need to look at the data to train its parameters? | Yes |
+    /// | Input timestamp column data type | <xref:System.DateTime> |
```

Comment:
For other time series functionality in ML.NET, there is no DateTime parameter (only a sequence of doubles)? I think there are assumptions about data points being equidistant still right?